src/components/StakingVaults/DeFiEarn.tsx

```diff
@@ -29,7 +30,8 @@ export const DeFiEarn: React.FC<DefiEarnProps> = ({
   header,
   ...rest
 }) => {
-  const [searchQuery, setSearchQuery] = useState('')
+  const { q } = useQuery<{ q: string }>()
+  const [searchQuery, setSearchQuery] = useState(q ?? '')
```

Comment:
`q` will always be defined according to types

Reply:
ah good catch!